pdns/misc.cc

```diff
@@ -1311,3 +1310,28 @@ unsigned int pdns_stou(const std::string& str, size_t * idx, int base)
   return static_cast<unsigned int>(result);
 }
 
+#if HAVE_OPENSSL_LIBCRYPTO
+#include <openssl/opensslv.h>
+#include <openssl/crypto.h>
+#endif
+
+bool constantTimeStringEquals(const std::string& a, const std::string& b)
+{
+  if (a.size() != b.size()) {
+    return false;
```

Comment:
You are right! That didn't matter until now because this function was only used to compare strings of the same length, and I haven't taken the time to check it before reusing it, my bad. It's a bit different in `C++11` because `std::string::size()` is mandated to be constant-time, but we leak information about the length nonetheless. I guess I'll have to rewrite that function if I ever find the time to do so.